liquibase-standard/src/test/groovy/liquibase/changelog/DatabaseChangeLogTest.groovy

```diff
@@ -702,4 +712,39 @@ http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbch
         0        | Integer.MAX_VALUE | 4
     }
 
+    @Unroll
+    def "includeAll finds all expected changelogs with EndsWithFilter: #endsWithFilter"() {
+        when:
+        def rootChangeLogPath = "com/example/root.xml"
+        def includedAllChangeLogPath = "changelogs"
+        def resourceAccessor = new MockResourceAccessor(["com/example/root.xml": "",
+                                                         "changelogs/changelog-1.xml": test3Xml,
+                                                         "changelogs/morechangelogs/changelog-2.xml": test3Xml,
+                                                         "changelogs/morechangelogs/withMore/changelog-3.xml": test3Xml,
+                                                         "changelogs/morechangelogs/withMore/changelog-4.xml": test3Xml,
+                                                         "changelogs/morechangelogs/AndMore/changelog-4.xml": test3Xml])
+
+        def rootChangeLog = new DatabaseChangeLog(rootChangeLogPath)
+        rootChangeLog.load(new ParsedNode(null, "databaseChangeLog")
+                .addChildren([includeAll: [path: includedAllChangeLogPath, endsWithFilter:endsWithFilter, errorIfMissing:false]]), resourceAccessor)
+
+        then:
+        rootChangeLog.getChangeSets().size() == expectedIncludeAllChangesetsToDeploy
+
+        where:
+        endsWithFilter  | expectedIncludeAllChangesetsToDeploy
+        null            | 5
+        ""              | 5
+        "1.XML"         | 1
+        "2.XML"         | 1
+        "3.XML"         | 1
+        "4.XML"         | 2
+        "5.XML"         | 0
```

Comment:
The test looks good to me, but it's failing because the negative scenarios are not able to find a file. I think we could create a separate test for it and make sure an exception is thrown, but this is already implemented (see test `includeAll throws an exception when directory not found`).

Reply:
Sorry, I could have sword I tested this and it had all the test passing, my bad! Regardless,  I would have thought including the `errorIfMissing:false` would take care of that? Would you agree that if the filter provides 0 changesets to execute, no error should be thrown if `errorIfMissing` is set to `false`? If you agree, I'm guessing I need to look into why its throwing that error and not respecting the `errorIfMissing` property

Reply:
Hmmm, I might need your help with this one @MalloD12. When I debugged that run, I noticed that the DatabaseChangeLog.java file is referencing the variable `errorIfMissingOrEmpty`, which you changed in merge #3901 to be `errorIfMissing`. When this is up to date with master, why would this not be present in this branch?

Appreciate any help here

Reply:
Oh, we undo that renaming last Friday for `includeAll` because `errorIfMissingOrEmpty` was still available in the xsd file and was causing confusion. It was a ship blocker a member of the team found on Friday afternoon (see #4433 for more details). So please, use `errorIfMissingOrEmpty`.

Reply:
AH, that explains everything! Okay, I was worried about that happening, glad to see it got undone just for includeAll. Also explains why I thought I had all the tests working: I did, before that change got pushed through! 🤣

I have updated the test so it uses the appropriate property name and now all the tests are passing